phpBB/styles/prosilver/template/posting_buttons.html

```diff
@@ -105,6 +105,16 @@
 	<button type="button" class="button button-icon-only bbcode-color" name="bbpalette" id="bbpalette" value="{L_FONT_COLOR}" onclick="change_palette();" title="{L_BBCODE_S_HELP}">
 		<i class="icon fa-tint fa-fw" aria-hidden="true"></i>
 	</button>
+	<!-- IF S_SMILIES_ALLOWED -->
+	<button type="button" class="button button-icon-only smiley_button" accesskey="y" name="addsmiley" title="{L_SMILEYS}">
```

Comment:
I have changed my code.

 <div id="smiley-dropdown" class="dropdown smiley-dropdown" >
     <div class="pointer"><div class="pointer-inner"></div></div>
     <ul class="dropdown-contents smiley-list" role="menu">
        <!-- EVENT posting_editor_smilies_before -->
        <!-- IF S_SMILIES_ALLOWED and .smiley -->
            <!-- BEGIN smiley -->
            <li class="smiley-list-item">
                <a class="smiley-add" data-smiley-text="{smiley.A_SMILEY_CODE}"><img class="smiley-image" src="{smiley.SMILEY_IMG}" width="{smiley.SMILEY_WIDTH}" height="{smiley.SMILEY_HEIGHT}" alt="{smiley.SMILEY_CODE}" title="{smiley.SMILEY_DESC}" /></a>
            </li>
            <!-- END smiley -->
        <!-- ENDIF -->
        <!-- EVENT posting_editor_smilies_after -->

        <!-- EVENT posting_editor_bbcode_status_after -->
        <!-- IF S_EDIT_DRAFT || S_DISPLAY_REVIEW -->
        <!-- IF S_DISPLAY_REVIEW --><hr /><!-- ENDIF -->
        <!-- IF S_EDIT_DRAFT --><strong><a href="{S_UCP_ACTION}">{L_BACK_TO_DRAFTS}</a></strong><!-- ENDIF -->
        <!-- IF S_DISPLAY_REVIEW --><strong><a href="#review">{L_TOPIC_REVIEW}</a></strong><!-- ENDIF -->
        <!-- ENDIF -->

        <!-- IF S_SHOW_SMILEY_LINK and S_SMILIES_ALLOWED -->
            <div class="footer">
                <a  href="{U_MORE_SMILIES}"><span>{L_MORE_SMILIES}</span></a>
            </div>
        <!-- ENDIF -->
    </ul>
 </div>

Do you think still it's not correct.